Gems/ScriptCanvasDeveloper/Code/Editor/Source/Tools/TranslationBrowser/TranslationBrowser.h

```diff
@@ -0,0 +1,384 @@
+        Q_OBJECT
+
+    public:
+
+        struct TreeNode
+        {
+            QString m_name;
+            QString m_type;
+            AZ::BehaviorClass* m_behaviorClass;
+            const AZ::SerializeContext::ClassData* m_classData;
+
+            TreeNode() {}
+            TreeNode(const TreeNode& rhs)
+            {
+                m_name = rhs.m_name;
+                m_type = rhs.m_type;
+                m_behaviorClass = rhs.m_behaviorClass;
+                m_classData = rhs.m_classData;
+            }
+            virtual ~TreeNode() {}
+
+            TreeNode(const QString& name, const QString& type, AZ::BehaviorClass* bcClass)
+                : m_name(name)
+                , m_type(type)
+                , m_behaviorClass(bcClass)
+                , m_classData(nullptr)
+            {}
+
+            TreeNode(const QString& name, const QString& type, const AZ::SerializeContext::ClassData* classData)
+                : m_name(name)
+                , m_type(type)
+                , m_behaviorClass(nullptr)
+                , m_classData(classData)
+            {}
+        };
+
+        const TreeNode* nodeForIndex(const QModelIndex& index) const
+        {
+            if (!index.isValid())
+            {
+                return nullptr;
+            }
+
+            Q_ASSERT(index.row() >= 0 && index.row() < static_cast<int>(m_topLevelItems.size()));
+            return m_topLevelItems[index.row()].get();
+        }
+
+        std::vector<std::shared_ptr<TreeNode>> m_topLevelItems;
+        size_t Count() const { return m_topLevelItems.size(); }
+
+        BehaviorClassModel();
+
+
+        void PopulateScriptCanvasNodes();
+
+        /*QModelIndex index(int row, int column, const QModelIndex& parent = QModelIndex()) const override
+        {
+            if (row >= 0 && row < m_classes.size() && column == 0)
+            {
+                return createIndex(row, column);
+            }
+
+            return {};
+        }
+
+        int rowCount(const QModelIndex& parent = QModelIndex()) const override
+        {
+            return aznumeric_cast<int>(m_topLevelItems.size());
+        }
+
+
+        int columnCount(const QModelIndex& parent = QModelIndex()) const override
+        {
+            return 1;
+        }
+
+
+        QVariant data(const QModelIndex& index, int role = Qt::DisplayRole) const override
+        {
+            AZ::BehaviorClass* bcClass = m_classes[index.row()];
+            if (bcClass)
+            {
+
+                switch (role)
+                {
+                case Qt::DisplayRole:
+                {
+                    switch (index.column())
+                    {
+                    case 0: return bcClass->m_name.c_str();
+                    }
+                    break;
+                }
+
+                }
+            }
+
+            return {};
+        }
+
+
+        QModelIndex parent(const QModelIndex& child) const override
+        {
+            return {};
+        }*/
```

Comment:
What is with the commented out functions?

Reply:
This tool will be removed, it was only used to migrate text from the old system to the new. I'll cut the entire file soon.